src/MongoDB.Driver.Core/Core/Operations/UpdateOpcodeOperation.cs

```diff
@@ -153,14 +152,17 @@ public WriteConcernResult Execute(IWriteBinding binding, CancellationToken cance
         /// <inheritdoc/>
         public WriteConcernResult Execute(RetryableWriteContext context, CancellationToken cancellationToken)
         {
-            if (Feature.WriteCommands.IsSupported(context.Channel.ConnectionDescription.ServerVersion) && _writeConcern.IsAcknowledged)
-            {
-                var emulator = CreateEmulator();
-                return emulator.Execute(context, cancellationToken);
-            }
-            else
+            using (EventContext.BeginOperation())
```

Comment:
I simply moved the call to EventContext.BeginOperation from the existing Execute method to the new overload so that it would be called no matter which Execute overload was used.